### PR TITLE
fix: [CDS-25770]: added option to view in new tab and preview yaml fo…

### PR DESCRIPTION
**Pull request by samarth-sharma31**
…r step and stage template

##### Summary:

Adds support to open template in new tab and also preview template yaml in a modal

##### Jira Links:

https://harness.atlassian.net/browse/CDS-25770

##### Screenshots:

Screenshot 2022-02-10 at 12 15 57 AM

Screenshot 2022-02-10 at 2 28 11 AM

https://user-images.githubusercontent.com/96409598/153344093-c575c06b-03ce-41f3-9074-db1fba1d8c0f.mov

https://user-images.githubusercontent.com/96409598/153404266-0f430f88-ab70-4bd6-b951-b7d0de6a0b5a.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/72-templates-library/components/TemplateStageSpecifications/TemplateStageSpecifications.tsx [hunk 1]
```diff
@@ -61,6 +61,7 @@ export const TemplateStageSpecifications = (): JSX.Element => {
   const queryParams = useParams<ProjectPathProps>()
   const { branch, repoIdentifier } = useQueryParams<GitQueryParams>()
   const [formValues, setFormValues] = React.useState<TemplateStageValues>(stage?.stage as TemplateStageValues)
+  // const [isTemplatePreviewModalOpen, setIsTemplatePreviewModalOpen] = React.useState(false)
```
Comment (yc147) [inline src/modules/72-templates-library/components/TemplateStageSpecifications/TemplateStageSpecifications.tsx:64]: Revert this change

File: src/modules/70-pipeline/components/PipelineStudio/TemplateBar/TemplateBar.tsx [hunk 2]
```diff
@@ -6,18 +6,21 @@
...
 import { getTemplateNameWithLabel } from '@pipeline/utils/templateUtils'
 import { useFeature } from '@common/hooks/useFeatures'
 import { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
-import type { ProjectPathProps, GitQueryParams } from '@common/interfaces/RouteInterfaces'
+import routes from '@common/RouteDefinitions'
+import type { PipelineType, ProjectPathProps, GitQueryParams } from '@common/interfaces/RouteInterfaces'
 import { useGetTemplate } from 'services/template-ng'
 import { getIdentifierFromValue, getScopeFromValue } from '@common/components/EntityReference/EntityReference'
+// eslint-disable-next-line no-restricted-imports
+import { TemplateYaml } from '@templates-library/components/TemplateYaml/TemplateYaml'
```
Comment (yc147) [inline src/modules/70-pipeline/components/PipelineStudio/TemplateBar/TemplateBar.tsx:23]: This is not the right way. Move TemplateYaml component to 70-pipeline and use it.

Comment (samarth-sharma31) [reply src/modules/70-pipeline/components/PipelineStudio/TemplateBar/TemplateBar.tsx]: Higher numbered modules can import from lower numbered modules but not the other way round to avoid cyclic dependency, got it!
Changes done! Thanks!

**Review by yc147 — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Review by yc147 — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]
